src/components/Swap/Speedbump/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Trans } from '@lingui/macro'
+import Button, { TextButton } from 'components/Button'
+import Column from 'components/Column'
+import { OnCloseDialogContext } from 'components/Dialog'
+import Row from 'components/Row'
+import { PriceImpact } from 'hooks/usePriceImpact'
+import { LargeAlert, LargeIcon, X } from 'icons'
+import { PropsWithChildren, useContext } from 'react'
+import styled, { css } from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+const SpeedbumpWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  padding: 1em;
+  text-align: center;
+`
+
+const Body = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  justify-content: space-between;
+`
+
+const BodyText = styled(ThemedText.Body1)`
+  padding: 0 0.5em;
+`
+
+const IconWrapper = styled.div`
+  padding: 2em;
+`
+const SpeedbumpButtonStyle = css`
+  border-radius: 1em;
+  padding: 1em;
+`
+const HeaderRow = styled(Row)`
+  align-items: flex-start;
+  width: 100%;
+`
+
+const StyledXButton = styled(LargeIcon).attrs({ icon: X, color: 'primary', size: 1.5 })`
+  :hover {
+    cursor: pointer;
+  }
+`
+const ContinueButton = styled(Button)`
+  ${SpeedbumpButtonStyle}
+  background-color: ${({ theme }) => theme.criticalSoft};
+  color: ${({ theme }) => theme.critical};
+`
+const CancelButton = styled(TextButton)`
+  ${SpeedbumpButtonStyle}
+  color: ${({ theme }) => theme.secondary};
+`
```

Comment:
do you think we should be using `ActionButton` for these? (and adding to its logic for bg/text colors) ?

Reply:
I'd rather not in this PR, that would involve changing both ActionButton and the BaseButton in Button.tsx to allow overriding colors. Action button also has lots of complicated features that this button doesn't need